dandi/dandiapi.py

```diff
@@ -971,8 +971,22 @@ def get_metadata(self) -> models.Dandiset:
         """
         Fetch the metadata for this version of the Dandiset as a
         `dandischema.models.Dandiset` instance
+
+        .. note::
+
+            Only published Dandiset versions can be expected to have valid
+            metadata.  Consider using `get_raw_metadata()` instead in order to
+            fetch unstructured, possibly-invalid metadata.
         """
-        return models.Dandiset.parse_obj(self.get_raw_metadata())
+        try:
+            return models.Dandiset.parse_obj(self.get_raw_metadata())
+        except ValidationError as e:
+            raise ValueError(
```

Comment:
IMHO we should not adapt into another type of exception here -- [ValidationError](https://github.com/pydantic/pydantic/blob/HEAD/pydantic/v1/error_wrappers.py#L50) has `.errors()` etc...

could you please subclass it locally with `__init__` to copy over errors and model and adjust its `__str__` to get that note ?

Reply:
While subclassing looks like it would work in Pydantic 1.x, it won't work in 2.x, as there `ValidationError` is implemented in Rust, and the only public Python method I see for creating a `ValidationError` is a static method that can only ever return a `ValidationError`, not a custom subclass.

Reply:
oh! I have missed `**v1**` in that `pydantic/v1/error_wrappers.py` path... what could possibly go wrong by mixing Python and Rust? I guess we are yet to see and for now just enjoy public classes/interfaces like

```
In [5]: ValidationError?
Init signature: ValidationError(self, /, *args, **kwargs)
Docstring:      Inappropriate argument value (of correct type).
File:           ~/venvs/dev3.11/lib/python3.11/site-packages/pydantic_core/_pydantic_core.cpython-311-x86_64-linux-gnu.so
Type:           type
Subclasses:   
```

which cannot even

```
In [6]: ValidationError()
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[6], line 1
----> 1 ValidationError()

TypeError: No constructor defined
```

heh...  I really do not see then a way we could improve UX/DX here then... we should not issue warnings etc. I guess let's just leave the `note` in the doc string and be done with it.